### ISSUE TITLE
Build failure on gcc8

### ISSUE THREAD

**Issue by amreo** [colobot/colobot#1180]
Hi. I've tryied to setup colobot from AUR (archlinux) but it has failed in building stage. Which maybe is the error?

 `[ 45%] Building CXX object src/CMakeFiles/colobotbase.dir/graphics/model/model_output.cpp.o
/home/amreo/temp/yaourt-tmp-amreo/aur-colobot-gold/src/colobot-colobot-gold-0.1.11.1-alpha/src/graphics/model/model_output.cpp: In function ‘void Gfx::ModelOutput::WriteOldModel(const Gfx::CModel&, std::ostream&)’:
/home/amreo/temp/yaourt-tmp-amreo/aur-colobot-gold/src/colobot-colobot-gold-0.1.11.1-alpha/src/graphics/model/model_output.cpp:331:16: error: ‘char* strncpy(char*, const char*, size_t)’ specified bound 20 equals destination size [-Werror=stringop-truncation]
         strncpy(t.texName, triangle.tex1Name.c_str(), 20);
         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [src/CMakeFiles/colobotbase.dir/build.make:752: src/CMakeFiles/colobotbase.dir/graphics/model/model_output.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:219: src/CMakeFiles/colobotbase.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
`

**Comment by Erihel** [comment]
What compiler do you use? Do you have some compiler flags set in `/etc/makepkg.conf` that treat warnings as errors?

This warning comes from code that's marked as deprecated. Quick workaround would be to change 20 to 19 in `src/graphics/model/model_output.cpp:331`

**Comment by amreo** [comment]
I'm using GCC 8.1.1 on GNU/Linux. These are the default flags present in /etc/makepkg
`#########################################################################
# ARCHITECTURE, COMPILE FLAGS
#########################################################################
#
CARCH="x86_64"
CHOST="x86_64-pc-linux-gnu"

#-- Compiler and Linker Flags
CPPFLAGS="-D_FORTIFY_SOURCE=2"
CFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt"
CXXFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt"
LDFLAGS="-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now"
#-- Make Flags: change this for DistCC/SMP systems
#MAKEFLAGS="-j2"
#-- Debugging flags
DEBUG_CFLAGS="-g -fvar-tracking-assignments"
DEBUG_CXXFLAGS="-g -fvar-tracking-assignments"
`

**Comment by melex750** [comment]
Maybe add this: ```-Wno-stringop-truncation```

**Comment by Erihel** [comment]
There was some change as it started to fail for me as well.

@tomangelo2 Setting `-DTOOLS=0` will not help as this file is listed in main CMakeLists under `BASE_SOURCES`.

Anyway that's not the only problem. There are currently some other problems if `-Werror` is set under gcc version 8.1.1

```
src/ui/screen/screen_io.cpp:84:19: error: ‘ - ’ directive writing 3 bytes into a region of size between 1 and 100 [-Werror=format-overflow=]
     sprintf(name, "%s - %s %d", line, resume.c_str(), m_main->GetLevelRank());
                   ^~~~~~~~~~~~
```

**Comment by krzys-h** [comment]
This particular warning was fixed in #1192 but there are more, see #1195

**Comment by MrSimbax** [comment]
Fixed in #1199
